src/main/scala/gemmini/LoadController.scala

```diff
@@ -7,14 +7,17 @@ import Util._
 import freechips.rocketchip.config.Parameters
 
 // TODO deal with errors when reading scratchpad responses
-class LoadController[T <: Data, U <: Data, V <: Data](config: GemminiArrayConfig[T, U, V], coreMaxAddrBits: Int, local_addr_t: LocalAddr)
+class LoadController[T <: Data, U <: Data, V <: Data](config: GemminiArrayConfig[T, U, V], coreMaxAddrBits: Int, local_addr_t: LocalAddr, num_dma: Int)
                                (implicit p: Parameters) extends Module {
   import config._
 
   val io = IO(new Bundle {
     val cmd = Flipped(Decoupled(new GemminiCmd(rob_entries)))
 
-    val dma = new ScratchpadReadMemIO(local_addr_t, mvin_scale_t_bits)
+    //val dma = new ScratchpadReadMemIO(local_addr_t, mvin_scale_t_bits)
+    //val dma_A = new ScratchpadReadMemIO(local_addr_t, mvin_scale_t_bits)
+    //val dma_B = new ScratchpadReadMemIO(local_addr_t, mvin_scale_t_bits)
+    val dma = Seq.fill(num_dma)(Module(new ScratchpadReadMemIO(local_addr_t, mvin_scale_t_bits)))
 
     val completed = Decoupled(UInt(log2Up(rob_entries).W))
```

Comment:
I think its better to place this arbiter outside the load controller,to avoid having to change logic in here